### PR TITLE
ced recording

### PR DESCRIPTION
**Pull request by luiztauffer**
- exclude smrx_channel_ids from source_schema
- fix file_path dictionary format for source_schema
- fix BaseRecordingExtractorInterface.get_metadata_schema()
- commented out fill_defaults

@bendichter I've been having problems with the `fill_defaults` function, and never really needed to use it. Is it important?

this PR corrects the source_schema for CEDRecordingInterface, but maybe not in the best way possible... I'm manually adding info that should come automatically from `get_schema_from_method_signature`, but not sure where to add the control for "file" or "directory" for example

### CODE REVIEW

**Comment by bendichter** [comment]
@luiztauffer what problems have you been having with fill_defaults?

**Comment by luiztauffer** [comment]
> @luiztauffer what problems have you been having with fill_defaults?

@bendichter I think it got fixed with the fixing of metadata_schema for BaseRecordingExtractorInterface. I uncommented it now

**Comment by luiztauffer** [comment]
@CodyCBakerPhD 
this changed `BaseRecordingExtractorInterface.get_metadata_schema()`, which was incorrect before. Please check it, it might be relevant for your converters

File: nwb_conversion_tools/datainterfaces/ceddatainterface.py [hunk 1]
```diff
@@ -13,7 +13,20 @@ class CEDRecordingInterface(BaseRecordingExtractorInterface):
     @classmethod
     def get_source_schema(cls):
         """Compile input schema for the RecordingExtractor."""
-        return get_schema_from_method_signature(
+        source_schema = get_schema_from_method_signature(
             class_method=cls.RX.__init__,
-            exclude=['smrx_ch_inds']
+            exclude=['smrx_channel_ids']
         )
+        source_schema.update(additionalProperties=True)
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ceddatainterface.py:20]: Why `additionalProperties=True`?

File: nwb_conversion_tools/datainterfaces/ceddatainterface.py [hunk 2]
```diff
@@ -13,7 +13,20 @@ class CEDRecordingInterface(BaseRecordingExtractorInterface):
...
+        source_schema['properties'].update(
+            file_path=dict(
+                type=source_schema['properties']['file_path']['type'],
+                format="file",
+                description="path to data file"
+            )
+        )
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ceddatainterface.py:27]:
```suggestion
        source_schema['properties']['file_path'].update(
            format="file",
            description="path to data file"
        )
```

would something like this work?

File: nwb_conversion_tools/datainterfaces/ceddatainterface.py [hunk 3]
```diff
@@ -13,7 +13,20 @@ class CEDRecordingInterface(BaseRecordingExtractorInterface):
...
+        return source_schema
+
+    @classmethod
+    def get_all_channels_info(cls, file_path):
+        return cls.RX.get_all_channels_info(file_path)
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ceddatainterface.py:32]: It seems like this method might be better in a higher class.

File: nwb_conversion_tools/datainterfaces/ceddatainterface.py (hunk 1 above)
Comment (luiztauffer) [reply nwb_conversion_tools/datainterfaces/ceddatainterface.py:20]: bacause of `smrx_channel_ids`, that is not present on schema

File: nwb_conversion_tools/datainterfaces/ceddatainterface.py (hunk 3 above)
Comment (luiztauffer) [reply nwb_conversion_tools/datainterfaces/ceddatainterface.py:32]: it is specific to CED though

**Comment by CodyCBakerPhD** [comment]
@luiztauffer @bendichter My bad, I was adding some changes to this branch thinking it was one of mine and just now realized. Reverted all changes, opening them on other PR.

@luiztauffer The `get_metadata_schema()` is really only used for your forms, I don't enforce it yet though it is on the to-do list to add something like that within the `spikeextractors` checks to reduce the amount of code over there.

**Comment by luiztauffer** [comment]
I rebased and solved the conflicts, it should be good to merge now

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
@luiztauffer I still show merge conflicts here

**Pull request merged**